src/engine.rs

```diff
@@ -92,6 +98,19 @@ where
             listeners.clone(),
         );
 
+        let (tx, rx) = mpsc::channel();
+        let stats_clone = stats.clone();
+        let memtables_clone = memtables.clone();
+        let metrics_flusher = ThreadBuilder::new()
+            .name("raft-engine-metrics".into())
+            .spawn(move || loop {
```

Comment:
let the user to call flush_metrics instead of doing itself?

Reply:
Okay.